plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/DBNUtils.java

```diff
@@ -142,6 +143,10 @@ private static void sortNodes(DBNNode[] children) {
             return;
         }
 
+        if (firstChild instanceof DBNDatabaseItem item && item.getObject() instanceof DBSTableIndexColumn) {
```

Comment:
Maybe DBSTableIndexColumn -> DBPObjectWithOrdinalPosition ?
We also don't want to order constrain columns